src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -36,7 +36,11 @@ public HeapElement(int size, Heap heap)
         {
             allocated = Count = size;
             this.heap = heap;
-            values = Enumerable.Repeat(StackValue.BuildInvalid(), allocated).ToArray();
+            if (size != 0)
+            {
+                values = Enumerable.Repeat(StackValue.BuildInvalid(), allocated).ToArray();
```

Comment:
if we are worrying about performance - is this worth looking at replacing with a for loop?

Reply:
True... I can make the change